script/tasks/diamondSyncSigs_FAST.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+function diamondSyncSigs_FAST {
+  echo ""
+  echo "[info] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> running script syncSIGs now...."
+
+  # Load environment variables
+  source .env
+
+  # Load configuration and helper functions
+  source script/helperFunctions.sh
+
+  # Read function arguments
+  local NETWORK="$1"
+  local ENVIRONMENT="$2"
+  local DIAMOND_CONTRACT_NAME="$3"
+  local EXIT_ON_ERROR="$4"
+
+  # Set max number of concurrent jobs
+  local MAX_CONCURRENT_JOBS=5
+
+  # Validate required config path
+  checkNetworksJsonFilePath || checkFailure $? "retrieve NETWORKS_JSON_FILE_PATH"
+
+  # no need to distinguish between mutable and immutable anymore
+  DIAMOND_CONTRACT_NAME="LiFiDiamond"
+
+  # if no NETWORK was passed to this function, ask user to select it
+  if [[ -z "$NETWORK" ]]; then
+    # find out if script should be executed for one network or for all networks
+    checkNetworksJsonFilePath || checkFailure $? "retrieve NETWORKS_JSON_FILE_PATH"
+    echo ""
+    echo "Should the script be executed on one network or all networks?"
+    NETWORK=$(echo -e "All (non-excluded) Networks\n$(jq -r 'keys[]' "$NETWORKS_JSON_FILE_PATH")" | gum filter --placeholder "Network")
+    echo "[info] selected network: $NETWORK"
+
+    if [[ "$NETWORK" != "All (non-excluded) Networks" ]]; then
+      checkRequiredVariablesInDotEnv $NETWORK
+    fi
+  fi
+
+  # create array with network/s for which the script should be executed
+  if [[ "$NETWORK" == "All (non-excluded) Networks" ]]; then
+    # get array with all network names
+    NETWORKS=($(getIncludedNetworksArray))
+  else
+    NETWORKS=($NETWORK)
+  fi
+
+  # Determine file suffix based on environment
+  FILE_SUFFIX=$(getFileSuffix "$ENVIRONMENT")
+
+  # Temp file to track any failed network logs from background jobs
+  FAILED_LOG_FILE=$(mktemp)
+
+  # Define function to process a single network
+  function processNetwork {
+    local NETWORK=$1
+
+    # Skip local/test networks
+    if [[ "$NETWORK" == "localanvil" || "$NETWORK" == "bsc-testnet" || "$NETWORK" == "lineatest" || "$NETWORK" == "mumbai" || "$NETWORK" == "sepolia" ]]; then
+      return
+    fi
```

Comment:
I think you can simplify this
```bash
local SKIP_NETWORKS=("localanvil" "bsc-testnet" "lineatest" "mumbai" "sepolia")
  if [[ " ${SKIP_NETWORKS[*]} " =~ " $NETWORK " ]]; then
    return
  fi
```

Reply:
Good point. Thank you. 
I went a bit further and introduced a helper function that checks if a network is an active mainnet, otherwise it skips it. 
We just need to that keep in mind for whenever we want to sync a network that is a testnet or not active (should be a very rare case imho).